### DISCUSSION TITLE
Counting with a sonoff basic..

### DISCUSSION THREAD

**Discussion by HelgeRoehrbein** [arendst/Tasmota#15376] [Q&A] [answered]
Hello everyone,
i want to count an impuls, made by an reed contact. Now i hope, that i can use a sonoff basic (tasmota) with a "counter" config.
But how to contact to the basic. 
i am an early beginner.., an excuse my bad english..

greetings
Helge

**Comment by barbudor** [comment, marked as answer]
Do you need the relay control of the Basic ?
If not, you'd better use a development board, running at a low voltage (5V/3.3V) which is safer than tickering an old product which is not really intended for that.
Many board such as NodeMCUor Wemos Mini-D1 are available, either standard version using ESP8266 or more recent using ESP32 or ESP32-C3.

Now if you insist on using the Basic, there is no easy connector so you'll have to open the box and solder on the connector:
https://tasmota.github.io/docs/devices/Sonoff-Basic/

Basically you could connect your reed switch between the GND pin and the RX Pin (==GPIO3) and configure GPIO3 as your counter input.
Because the input as an internal pull-up, when the reed switch is closed it will connect to GND and make a logical 0, and when the reed is open, it will make a logical 1

If you cable distances over 30-50cm, I would advise to have an external pull resistor of 1Kohm (or less if long distance) between 3.3V and RX pin

**Reply by HelgeRoehrbein** [reply]
Thanks a lot. I've now improvised with a jumper wire, but the counter jumps and counts more than one pulse. Do you have a tip for me?

**Reply by HelgeRoehrbein** [reply]
i do not need a relay control. i only want to count the impulse and after that to show how much impuls a day, sending with MQTT to my home assistant.

**Reply by barbudor** [reply]
You should look into the CounterDebounce command
And Google to understand what is bouncing and debouncing

**Comment by lalo-uy** [comment]
Put a pullup resistor of 1k between input an vcc 3.3v
Also a 100nF capacitor between inpur and ground.
Mutch safer if you switch to a development board wemod D1 mini or similar.
